src/v0/destinations/bluecore/utils.js

```diff
@@ -0,0 +1,344 @@
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { isObject } = require('../../util');
+const {
+  ACTION_SOURCES_VALUES,
+  CONFIG_CATEGORIES,
+  OTHER_STANDARD_EVENTS,
+  FB_PIXEL_CUSTOM_DATA_EXCLUDE_FLATTENING,
+} = require('./config');
+const { getContentType, getContentCategory } = require('../../util/facebookUtils');
+
+/**  format revenue according to fb standards with max two decimal places.
+ * @param revenue
+ * @return number
+ */
+
+const formatRevenue = (revenue) => {
+  const formattedRevenue = parseFloat(parseFloat(revenue || '0').toFixed(2));
+  if (!Number.isNaN(formattedRevenue)) {
+    return formattedRevenue;
+  }
+  throw new InstrumentationError('Revenue could not be converted to number');
+};
+
+/**
+ * Returns action source
+ * ref : https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/server-event#action-source
+ * @param {*} payload
+ * @param {*} channel
+ * @returns
+ */
+const getActionSource = (payload, channel) => {
+  let actionSource = 'other';
+  if (payload.action_source) {
+    const isActionSourceValid = ACTION_SOURCES_VALUES.includes(payload.action_source);
+    if (!isActionSourceValid) {
+      throw new InstrumentationError('Invalid Action Source type');
+    }
+    actionSource = payload.action_source;
+  } else if (channel === 'web') {
+    actionSource = 'website';
+  } else if (channel === 'mobile') {
+    actionSource = 'app';
+  }
+
+  return actionSource;
+};
+
+/**
+ *
+ * @param {*} message Rudder element
+ * @param {*} categoryToContent example: [ { from: 'clothing', to: 'product' } ]
+ *
+ * Handles order completed and checkout started types of specific events
+ */
+const handleOrder = (message, categoryToContent) => {
```

Comment:
this implementation is specific to facebook, please edit according to bluecore needs